quic/s2n-quic/src/tests.rs

```diff
@@ -370,6 +374,15 @@ event_recorder!(
         storage.push(event.pto_count);
     }
 );
+event_recorder!(
+    FrameReceivedRecorder,
+    FrameReceived,
+    on_frame_received,
+    Frame,
+    |event: &FrameReceived, storage: &mut Vec<Frame>| {
+        storage.push(event.frame.clone());
```

Comment:
I think the test would be a bit more interesting to track the handshake status events rather that frames sent/recv

Reply:
yea you are right. I wanted to enforce that the server also sent the HandshakeDone frame but we will capture that by asserting Handshake Confirmed on the client.

Reply:
Those events also give really good insight into the internal state of the endpoint